core/Archive/ArchiveInvalidator.php

```diff
@@ -528,20 +535,48 @@ public function removeInvalidations($idSite, $plugin, $report = null)
     }
 
     /**
-     * Re-archives reports without propagating exceptions.
+     * Schedules a re-archiving reports without propagating exceptions. This is scheduled
+     * since adding invalidations can take a long time and delay UI response times.
      *
      * @param int|int[]|'all' $idSites
      * @param string $pluginName
      * @param string|null $report
      * @param Date|null $startDate
      */
-    public function reArchiveReportSafely($idSites, string $pluginName, string $report = null, Date $startDate = null)
+    public function scheduleReArchiving($idSites, string $pluginName, string $report = null, Date $startDate = null)
     {
         try {
-            $this->reArchiveReport($idSites, $pluginName, $report, $startDate);
+            $reArchiveList = new ReArchiveList($this->logger);
+            $reArchiveList->add([$idSites, $pluginName, $report, $startDate ? $startDate->getTimestamp() : null]);
         } catch (\Throwable $ex) {
-            $logger = StaticContainer::get(LoggerInterface::class);
-            $logger->info("Failed to schedule rearchiving of past reports for $pluginName plugin.");
+            $this->logger->info("Failed to schedule rearchiving of past reports for $pluginName plugin.");
+        }
+    }
+
+    /**
+     * Applies the queued archiving rearchiving entries.
+     */
+    public function applyScheduledReArchiving()
+    {
+        $reArchiveList = new ReArchiveList($this->logger);
+        $items = $reArchiveList->getAll();
+
+        foreach ($items as $entry) {
+            try {
+                list($idSites, $pluginName, $report, $startDateTs) = $entry;
+
+                $this->reArchiveReport($idSites, $pluginName, $report, Date::factory($startDateTs));
```

Comment:
maybe we should only create date instance if startDateTs is not empty cause otherwise it might result in an error or creates a 'now' date or something else unexpected?

Reply:
👍